### DISCUSSION TITLE
[✨] manually preload a new route when you know you'll need it

### DISCUSSION THREAD

**Discussion by riencoertjens** [QwikDev/qwik-evolution#134] [Proposals For Qwik]
### Is your feature request related to a problem?

[discord thread](https://discord.com/channels/842438759945601056/1156560274329964555/1156560274329964555)

### Describe the solution you'd like

I have this page, where I have an animation that I want to complete before doing the navigation, it would be nice if there was a way to already start preloading the next page
```jsx
const Page = component$(() => {
  const submitting = useSignal(false);
  const navigate = useNavigation();

  const onChange = $((ev, el) => {
    submitting.value = true;
    const formData = new FormData(el);
    const url = new URL("/next-step");
    url.searchParams.append("some-value", formData.get("some-value"));
    setTimeout(() => navigate(url.toString()), 1000);
  });

  return (
    <form onChange$={onChange}>
      <input type="text" class={submitting.value ? "flicker-input-navigation-to-indicate-submission" : ""} />
    </form>
  );
});
```

### Describe alternatives you've considered

It would be nice if I could do something this to 
```jsx
preload(url)
setTimeout(() => navigate(url.toString()), 1000);
```
sort of what happens when you hover over a link

### Additional context

additionally and a more broader but similar feature would be to have different prefetch behaviour like in [remix](https://remix.run/docs/en/main/components/link#prefetch):

> `prefetch` Defines the data and module prefetching behavior for the link.
> 
> ```jsx
> <>
>   <Link /> {/* defaults to "none" */}
>   <Link prefetch="none" />
>   <Link prefetch="intent" />
>   <Link prefetch="render" />
>   <Link prefetch="viewport" />
> </>
> ```
> `none` - default, no prefetching
> `intent` - prefetches when the user hovers or focuses the link
> `render` - prefetches when the link renders
> `viewport` - prefetches when the link is in the viewport, very useful for mobile

**Comment by samyung0** [comment]
Manual prefetching should be possible by exposing loadClientData function in use-endpoint.ts
But there is a catch I can think of: if the user prefetches a route while authenticated, qwik-city will load the relevant q-data file with all the auth details, then if the user logouts without refreshing the page, then the user can visit the route still as an authed user since prefetching does not occur again.

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.
